### DISCUSSION TITLE
macOS camera

### DISCUSSION THREAD

**Discussion by JoseEstevesPt** [airsdk/Adobe-Runtime-Support#1645] [Q&A]
Something that is quite simple in windows is becoming an unsolved nightmare in macOS.
Using Animate, Air SDK 33.1.1.743 access web cameras... can anyone help...?
Where can I find a working example?

thanks

**Comment by ajwfrost** [comment]
Animate doesn't have the permissions required to access webcams, but you should find it works if you're able to launch your application via a terminal window using `adl` - it should ask for permission first.

There's a long thread under https://github.com/airsdk/Adobe-Runtime-Support/issues/972 which may have some further hints/info.. essentially though, it's macOS security making it difficult particularly for an IDE trying to launch a process that requires more permissions than the IDE has....

thanks

**Reply by JoseEstevesPt** [reply]
thanks for the reply.... but its a pain, you cant use the air debug launcher / simulator....can this be fixed?

**Reply by ajwfrost** [reply]
Yes we're looking to create a new AIR app that would need to be run on your computer prior to debugging: the normal 'ADL' would then see that it's being run from Animate/etc, and would send a message to this other AIR app to load up your app for debugging. So it would then run under the other process, where we would be able to define the permissions..
Needs a couple of changes in the runtime though to allow the new app to spawn a new instance of itself just for the debug session...

**Reply by JoseEstevesPt** [reply]
Any updates on this issue?
